transport_nantes/topicblog/views.py

```diff
@@ -1080,9 +1080,44 @@ class TopicBlogPressSend(PermissionRequiredMixin, LoginRequiredMixin,
     success_url = "/"
 
 
+class TopicBlogPressIndex(ListView):
+    model = TopicBlogPress
+    template_name = 'topicblog/topicblogpress_index.html'
+    context_object_name = 'press_releases_list'
+    paginate_by = 10
+
+    def get_queryset(self) -> list:
+
+        # Get the list of slugs that have been published at least once
+        all_published_slugs = (
+            TopicBlogPress.objects.filter(publication_date__isnull=False)
+            .values_list('slug', flat=True)
+            .distinct()
+        )
+
+        # Get the ids of the most recently published press releases for each
+        # slug
+        all_published_press_releases = []
```

Comment:
This should happen server-side.  Yes, we are small enough that this will fit in memory with small transfer time.  But it's not good programming practice to leave time bombs like this.  It's like a programmer in 1980 assuming that two digits is good for a year, or a programmer more recently assuming that 32 bits is enough for a timestamp.  It works the day you write it, but it's predictably going to fail some day unless your software isn't used.

Reply:
It does happen in Django, so server-side. 
A list of integer is quite reasonable, especially since we are not likely to hit hundreds of millions of published items.

You are trying to optimize a non-issue here, in my opinion.
Yes, the code might not scale if you published millions of press releases, but before then, all this would probably change.

Reply:
> It does happen in Django, so server-side. A list of integer is quite reasonable, especially since we are not likely to hit hundreds of millions of published items.
> 
> You are trying to optimize a non-issue here, in my opinion. Yes, the code might not scale if you published millions of press releases, but before then, all this would probably change.

I'm trying to optimize a junior developer so that he grows and makes fewer mistakes, writes less code that flags him out as not understanding the finer points of his craft.